radio/src/thirdparty/libopenui/src/filechoice.cpp

```diff
@@ -68,10 +65,11 @@ bool FileChoice::openMenu()
     for (;;) {
       res = sdReadDir(&dir, &fno, firstTime);
       if (res != FR_OK || fno.fname[0] == 0)
-        break;                             // break on error or end of dir
-      if (fno.fattrib & AM_DIR) continue;  // skip subfolders
-      if (fno.fattrib & AM_HID) continue;  // skip hidden files
-      if (fno.fattrib & AM_SYS) continue;  // skip system files
+        break;  // break on error or end of dir
+      if (fno.fattrib & (AM_HID | AM_SYS | AM_DIR))
+        continue;  // Ignore subfolders, hidden files and system files
+      if (fno.fname[0] == '.' && fno.fname[1] != '.')
```

Comment:
Isn't there a missing length check? Or is it assuming that you would hit `\0` anyways?

Reply:
Code was copied from the file browser which has not had any problems that I know of. I will take a closer look though just in case.

Reply:
I think it's ok as is. The '..' file will pass through, everything else that starts with a dot gets filtered. The fno.fname buffer is 256 bytes so no overrun.